README.md

```diff
@@ -285,6 +285,68 @@ There are many places in the format for compatible extensions:
 - Encodings: Encodings are specified by enum and more can be added in the future.
 - Page types: Additional page types can be added and safely skipped.
 
+### Thrift extensions
+Thrift is used for metadata. The Thrift spec mandates that unknown fields are
+skipped. To facilitate extensions Parquet reserves field-id `32767` of *every*
+struct as an ignorable extension point. More specifically Parquet guarantees
+that field-id `32767` will *never* be seen in the official Thrift IDL. The type
+of this field is always `binary` for maximum extensibility and fast skipping by
```

Comment:
Should we make this `BYTE_ARRAY` as per https://github.com/apache/parquet-format/pull/251?
```suggestion
of this field is always `BYTE_ARRAY` for maximum extensibility and fast skipping by
```

Reply:
Thrift doesn't have ``BYTE_ARRAY`, just `binary`. The former is a physical type in Parquet, but thrift messages don't use Par quet types.